### PR TITLE
Make POW work on GTSs and Lists the same way numerical functions do

### PR DESCRIPTION
**Pull request by ftence**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/script/binary/POW.java [hunk 1]
```diff
@@ -16,35 +16,138 @@
...
+      if (!(gts1.getType() == GeoTimeSerie.TYPE.DOUBLE || gts1.getType() == GeoTimeSerie.TYPE.LONG) || !(gts2.getType() == GeoTimeSerie.TYPE.DOUBLE || gts2.getType() == GeoTimeSerie.TYPE.LONG)) {
+        throw new WarpScriptException(typeCheckErrorMsg);
+      }
+
+      // The result type is LONG if both inputs are LONG.
+      GeoTimeSerie result = new GeoTimeSerie(Math.max(GTSHelper.nvalues(gts1), GTSHelper.nvalues(gts2)));
+      GeoTimeSerie.TYPE type = (gts1.getType() == GeoTimeSerie.TYPE.LONG && gts2.getType() == GeoTimeSerie.TYPE.LONG) ? GeoTimeSerie.TYPE.LONG : GeoTimeSerie.TYPE.DOUBLE;
+      result.setType(type);
+
+      GTSOpsHelper.GTSBinaryOp op = new GTSOpsHelper.GTSBinaryOp() {
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/binary/POW.java:96]: This does not depend on the operands, can it be instantiated once and reused?

File: warp10/src/main/java/io/warp10/script/binary/POW.java [hunk 2]
```diff
@@ -16,35 +16,138 @@
...
+        result = result.cloneEmpty();
+      }
+
+      stack.push(result);
+    } else if ((op1 instanceof GeoTimeSerie && op2 instanceof Number) || (op1 instanceof Number && op2 instanceof GeoTimeSerie)) {
+      boolean op1gts = op1 instanceof GeoTimeSerie;
+
+      int n = op1gts ? GTSHelper.nvalues((GeoTimeSerie) op1) : GTSHelper.nvalues((GeoTimeSerie) op2);
+
+      GeoTimeSerie result = op1gts ? ((GeoTimeSerie) op1).cloneEmpty(n) : ((GeoTimeSerie) op2).cloneEmpty();
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/binary/POW.java:116]: missing hint if op2 is a GTS

File: warp10/src/main/java/io/warp10/script/binary/POW.java [hunk 3]
```diff
@@ -16,35 +16,138 @@
...
       }
+
+      if (!(gts.getType() == GeoTimeSerie.TYPE.LONG || gts.getType() == GeoTimeSerie.TYPE.DOUBLE)) {
+        throw new WarpScriptException(typeCheckErrorMsg);
+      }
+
+      Number op = op1gts ? (Number) op2 : (Number) op1;
+
+      // The result type is LONG if both inputs are LONG.
+      GeoTimeSerie.TYPE type = (gts.getType() == GeoTimeSerie.TYPE.LONG && op instanceof Long) ? GeoTimeSerie.TYPE.LONG : GeoTimeSerie.TYPE.DOUBLE;
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/script/binary/POW.java:132]: GeoTimeSerie.TYPE.LONG == gts.getType()

**Review by hbs — CHANGES_REQUESTED** [review]

**Comment by ftence** [comment]
Use a different approach.
Used in conjunction with #832, makes POW work on GTSs.

**Pull request merged**